src/Picasso_BatchedLinearAlgebra.hpp

```diff
@@ -1272,76 +1478,1389 @@ struct VectorView
+        operator-=( const Expression& e )
+    {
+        static_assert( Expression::extent_0 == extent_0, "Extents must match" );
+        static_assert( Expression::extent_1 == extent_1, "Extents must match" );
+        static_assert( Expression::extent_2 == extent_2, "Extents must match" );
+
+        for ( int i = 0; i < M; ++i )
+#if defined( KOKKOS_ENABLE_PRAGMA_UNROLL )
+#pragma unroll
+#endif
+            for ( int j = 0; j < N; ++j )
+            {
+                for ( int k = 0; k < P; ++k )
+                    ( *this )( i, j, k ) -= e( i, j, k );
+            }
+        return *this;
+    }
+
+    // Initializer list assignment operator.
+    KOKKOS_INLINE_FUNCTION
+    Tensor3& operator=( const std::initializer_list<
+                        std::initializer_list<std::initializer_list<T>>>
+                            data )
+    {
+        int i = 0;
+        int j = 0;
+        int k = 0;
+        for ( const auto& slice : data )
+        {
+            j = 0;
+            for ( const auto& row : slice )
+            {
+                k = 0;
+                for ( const auto& value : row )
+                {
+                    _d[i][j][k] = value;
+                    ++k;
+                }
+                ++j;
+            }
+            ++i;
+        }
+
+        return *this;
+    }
+
+    // Scalar value assignment.
+    KOKKOS_INLINE_FUNCTION
+    Tensor3& operator=( const T value )
+    {
+        for ( int i = 0; i < M; ++i )
+#if defined( KOKKOS_ENABLE_PRAGMA_UNROLL )
+#pragma unroll
+#endif
+            for ( int j = 0; j < N; ++j )
+            {
+                for ( int k = 0; k < P; ++k )
+                    ( *this )( i, j, k ) = value;
+            }
+        return *this;
+    }
+
+    // Strides.
+    KOKKOS_INLINE_FUNCTION
+    int stride_0() const { return N; }
+
+    KOKKOS_INLINE_FUNCTION
+    int stride_1() const { return P; }
+
+    KOKKOS_INLINE_FUNCTION
+    int stride_2() const { return 1; }
+
+    KOKKOS_INLINE_FUNCTION
+    int stride( const int d ) const
+    {
+        return ( 0 == d ) ? N : ( 1 == d ? P : 1 );
+    }
+
+    // Extent
+    KOKKOS_INLINE_FUNCTION
+    constexpr int extent( const int d ) const
+    {
+        return d == 2 ? extent_2
+                      : ( d == 0 ? extent_0 : ( d == 1 ? extent_1 : 0 ) );
+    }
+
+    // Access an individual element.
+    KOKKOS_INLINE_FUNCTION
+    const_reference operator()( const int i, const int j, const int k ) const
+    {
+        return _d[i][j][k];
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    reference operator()( const int i, const int j, const int k )
+    {
+        return _d[i][j][k];
+    }
+
+    // Get the raw data.
+    KOKKOS_INLINE_FUNCTION
+    pointer data() const { return const_cast<pointer>( &_d[0][0][0] ); }
+
+    // Get a row as a vector view.
+    KOKKOS_INLINE_FUNCTION
+    VectorView<T, M> vector( const Kokkos::Impl::ALL_t, const int n,
```

Comment:
@dalg24 what do you suggest here? We can create our own type instead, but using `Kokkos::ALL` seemed cleaner

Reply:
Now using `decltype(Kokkos::ALL)` to avoid direct use of Impl::